tests/python_package_test/test_basic.py

```diff
@@ -328,3 +281,77 @@ def check_asserts(data):
         lgb_data.set_weight(sequence)
         lgb_data.set_init_score(sequence)
         check_asserts(lgb_data)
+
+    def test_dataset_update_params_warning(self):
```

Comment:
could you either implement this same test on the R side or create an issue (to add to #2302 ) documenting the need for such a test on the R side?